### DISCUSSION TITLE
Unable to infer a hanlder when using mangum

### DISCUSSION THREAD

**Discussion by 94929** [Kludex/mangum#273] [Q&A]
Hi, I'm using mangum for serving lambda image and I get the following error:
```
{"errorMessage": "The adapter was unable to infer a handler to use for the event. This is likely related to how the Lambda function was invoked. (Are you testing locally? Make sure the request payload is valid for a supported handler.)", "errorType": "RuntimeError", "stackTrace": ["  File \"/var/lang/lib/python3.8/site-packages/mangum/adapter.py\", line 65, in __call__\n    handler = self.infer(event, context)\n", "  File \"/var/lang/lib/python3.8/site-packages/mangum/adapter.py\", line 55, in infer\n    raise RuntimeError(  # pragma: no cover\n"]}
```
Did anyone face the same issue? Any help would be appreciated.

**Comment by heisguyy** [comment]
I think the error should be with your request. What's your request like?

**Reply by 94929** [reply]
Thanks for the reply. This is the request I'm using:
```
curl -XPOST "http://localhost:3010/2015-03-31/functions/function/invocations" -d '{"count":1}' -H "Content-Type: application/json"
```

**Reply by heisguyy** [reply]
You are welcome. I had a very similar same problem when I tumbled on your question. 

```
curl -XPOST "http://localhost:3010/2015-03-31/functions/function/invocations" -d '{
        "resource": "/<your-endpoint-path>",
        "path": "/<your-endpoint-path>",
        "httpMethod": "POST",
        "headers": {"Content-Type": "application/json"}
        "requestContext": {},
        "body": '{"count":1}'
}'
```

Try this.

**Reply by heisguyy** [reply]
You have to mimick the structure of an event API gateway sends to a lambda proxy integration. You can see this structure in "Input format of a Lambda function for proxy integration" section of this  [article](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format)

**Reply by 94929** [reply]
Thanks, I now have set my request as below:
```
curl -XPOST "http://localhost:3010/2015-03-31/functions/function/invocations" -d '{
        "resource": "/resources",
        "path": "/resources",
        "httpMethod": "POST",
        "headers": {"Content-Type": "application/json"}
        "requestContext": {},
        "body": {"count":1}
}'
```

And I get the following:
```
{"errorMessage": "'dict' object has no attribute 'encode'", "errorType": "AttributeError", "stackTrace": ["  File \"/var/lang/lib/python3.8/site-packages/mangum/adapter.py\", line 71, in __call__\n    http_cycle = HTTPCycle(handler.scope, handler.body)\n", "  File \"/var/lang/lib/python3.8/site-packages/mangum/handlers/api_gateway.py\", line 82, in body\n    return maybe_encode_body(\n", "  File \"/var/lang/lib/python3.8/site-packages/mangum/handlers/utils.py\", line 23, in maybe_encode_body\n    body = body.encode()\n"]}
```

Did you face the same issue when making a request?

**Reply by heisguyy** [reply]
The value of the body should be in quotes

`"body": '{"count":1}'` instead of `"body":{"count":1}`

**Reply by 94929** [reply]
I did that in the first place, and got `{"errorMessage": "Unable to unmarshal input: Expecting value: line 1 column 1 (char 0)", "errorType": "Runtime.UnmarshalError", "stackTrace": []}` instead. So I thought the quotes for body is redundant as there already is quotes enclosing whole json on "-d".

**Reply by heisguyy** [reply]
Seems you are using a windows system. Check [this](https://stackoverflow.com/a/68091010)
